spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1899,22 +1904,21 @@ def run_script(self, filename, wdir, args, post_mortem, current_client,
                 method = "runfile"
             # If spyder-kernels, use runfile
             if client.shellwidget.is_spyder_kernel:
-                line = method + "('%s'" % (norm(filename))
+                line = "%{} {}".format(method , repr(norm(filename)))
                 if args:
-                    line += ", args='%s'" % norm(args)
-                if (
-                    wdir and client.shellwidget.is_external_kernel
-                    and os.path.samefile(wdir, os.path.dirname(filename))
-                ):
-                    # No working directory for external kernels
-                    wdir = ""
+                    line += ' --args {}'.format(repr(norm(args)))
                 if wdir:
-                    line += ", wdir='%s'" % norm(wdir)
+                    if wdir == os.path.dirname(filename):
+                        # No working directory for external kernels
+                        # If it has not been explicitely given
+                        if not client.shellwidget.is_external_kernel:
+                            line += " --wdir"
```

Comment:
So, now we simply pass `--wdir` to the `runfile` magic when `wdir` is the same directory where the file we're trying to execute is placed?

Reply:
Yes so this decreases the clutter in the console

Reply:
That's what I thought and it's a great idea! Well done 👍🏽